frame/ibc/src/lib.rs

```diff
@@ -345,118 +313,64 @@ pub mod pallet {
 			<T as Config>::AdminOrigin::ensure_origin(origin)?;
-			let mut ctx = routing::Context::<T>::new();
-			let type_url =
-				String::from_utf8(msg.type_url.clone()).map_err(|_| Error::<T>::DecodingError)?;
-			if type_url.as_str() != CREATE_CLIENT_TYPE_URL {
-				return Err(Error::<T>::InvalidMessageType.into())
-			}
-			let msg = ibc_proto::google::protobuf::Any { type_url, value: msg.value };
-
-			let MsgReceipt { events, log } =
-				ibc::core::ics26_routing::handler::deliver::<_, HostFunctions>(&mut ctx, msg)
-					.map_err(|_| Error::<T>::ProcessingError)?;
-
-			log::trace!(target: "pallet_ibc", "[pallet_ibc_deliver]: logs: {:?}", log);
-			Self::deposit_event(events.into());
-			Ok(())
-		}
 
-		#[pallet::weight(< T as Config >::WeightInfo::initiate_connection())]
-		#[frame_support::transactional]
-		pub fn initiate_connection(
-			origin: OriginFor<T>,
-			params: ConnectionParams,
-		) -> DispatchResult {
-			<T as Config>::AdminOrigin::ensure_origin(origin)?;
-			let client_id =
-				client_id_from_bytes(params.client_id).map_err(|_| Error::<T>::DecodingError)?;
-			if !ClientStates::<T>::contains_key(&client_id) {
-				return Err(Error::<T>::ClientStateNotFound.into())
-			}
-
-			let counterparty_client_id = client_id_from_bytes(params.counterparty_client_id)
-				.map_err(|_| Error::<T>::DecodingError)?;
-			let identifier = params.version.0;
-			let features = params.version.1;
-			let identifier =
-				String::from_utf8(identifier).map_err(|_| Error::<T>::DecodingError)?;
-			let features = features
-				.into_iter()
-				.map(String::from_utf8)
-				.collect::<Result<Vec<_>, _>>()
-				.map_err(|_| Error::<T>::DecodingError)?;
-			let raw_version =
-				ibc_proto::ibc::core::connection::v1::Version { identifier, features };
-			let version: Version = raw_version.try_into().map_err(|_| Error::<T>::DecodingError)?;
-
-			let commitment_prefix: CommitmentPrefix =
-				params.commitment_prefix.try_into().map_err(|_| Error::<T>::DecodingError)?;
-			let counterparty = Counterparty::new(counterparty_client_id, None, commitment_prefix);
-			let delay_period = core::time::Duration::from_nanos(params.delay_period);
-			let value = MsgConnectionOpenInit {
-				client_id,
-				counterparty,
-				version: Some(version),
-				delay_period,
-				signer: Signer::from_str(MODULE_ID).map_err(|_| Error::<T>::DecodingError)?,
-			}
-			.encode_vec();
-			let msg = ibc_proto::google::protobuf::Any {
-				type_url: CONNECTION_OPEN_INIT_TYPE_URL.to_string(),
-				value,
-			};
 			let mut ctx = routing::Context::<T>::new();
-			let result =
-				ibc::core::ics26_routing::handler::deliver::<_, HostFunctions>(&mut ctx, msg)
-					.map_err(|_| Error::<T>::ProcessingError)?;
-			Self::deposit_event(result.events.into());
+			let messages = messages
+				.into_iter()
+				.filter_map(|message| {
+					let type_url = String::from_utf8(message.type_url.clone()).ok()?;
+					let is_permissioned = matches!(
```

Comment:
why there's only need to have one permission attribute and not all?

Reply:
ref: https://github.com/ComposableFi/ibc-rs/issues/31

Reply:
I'm not arguing against verifying permissions. My question was regarding the permissions that were being verified. Seems that "as long as it has one" it's ok.

Reply:
It's a filter_map operation. So we filter out permissioned packets.

Reply:
got it, thanks

Reply:
I'll just rephrase my question, to be 100% sure:
here we are checking if the message matches _at least_ one permission. That's the definition of having the right to execute the message. However, couldn't it be that a message has rights for something but not for something else? and that we need to check in a more granular way?

Reply:
> here we are checking if the message matches at least one permission

That's actually not what is happening, we collect only packets that require root to be dispatched and dispatch them.

https://github.com/ComposableFi/composable/blob/4fe5537a274ca2186e821a6fd67f4443f6953662/frame/ibc/src/lib.rs#L352-L375